co-simulation tool/README.md

```diff
@@ -1,88 +1,248 @@
+	"carlaConnectionPort": 8913
+}
+```
 
+The following table descripts the parameters in the CARLA configuration file.
+
+| Parameters          | **Type** | **Description**                                              | **Required** |
+| ------------------- | -------- | ------------------------------------------------------------ | ------------ |
+| updateInterval      | number   | The Interval after which the CARLA is updated.  The default unit is ms. Define the size  of one simulation step in CARLA. The updateInterval of Carla and Sumo  ambassador must be the same. | Yes          |
+| carlaUE4Path        | string   | Path to CarlaUE4 executable file. If the environment  variable **`CARLA_HOME`** is configured to define  the installation directory of CarlaUE4 executable file, this parameter can be removed from  the configuration file. | No           |
+| bridgePath          | string   | This parameter includes two information separated by the  semicolon. The first substring is the path of bridge .sh or .bat. The second  substring is the file name of bridge .sh or .bat. The extension of the bridge file is needed. | Yes          |
+| carlaConnectionPort | number   | The bridge server port . It must be the same as  the bridge server port number in the bridge .sh or .bat file.  If the entry is not used, the default port number is 8913. | No           |
+
+**c.**  **Bridge file**
+
+The bridge file (bridge.sh or bridge.bat) is used to run the bridge to connect CARLA simulator and CARLA ambassador, load maps to CARLA simulator, pass messages between CARLA ambassador and CARLA simulator.  The TraCI protocol and Python API of CARLA simulator are used. The examples of  bridge files on Linux systems and on Windows systems are shown in the following, respectively.
+
+**Linux**
+
+```bash
+#!/bin/bash
+
+#x-terminal-emulator
+cd ../../../bridge
 
-[**Eclipse MOSAIC**](https://eclipse.org/mosaic) is a multi-scale simulation framework in the field of smart and connected mobility.
-It allows coupling simulators from various domains towards a comprehensive simulation tool. 
-Data exchange and time management is implemented by the Runtime Infrastructure (RTI), which is the heart of MOSAIC. Simulation models
-are coupled to the RTI using HLA inspired interfaces; Each simulator is wrapped into a "Federate" object which is linked to an "Ambassador"
-which is directly coupled with the RTI. Currently, the following simulators are coupled with the MOSAIC RTI:
-  * MOSAIC Application (application simulation)
-  * Eclipse SUMO (traffic simulation)
-  * OMNeT++ (communication simulation)
-  * ns-3 (communication simulation)
-  * MOSAIC Cell Simulator (cellular communication simulation)
-  * MOSAIC Simple Network Simulator (communication simulation)
-  * MOSAIC Environment (environment and event simulation)  
-  * MOSAIC Output Generator (evaluation and visualization)
+python3 carla_mosaic_bridge.py --bridge-server-port 8913 --map Town04 net/Town04.net.xml
```

Comment:
It is better to change the python3 to python3.7 in the bridge file under scenario folder because not all users set python 3.7 as default version in their system.

Reply:
The python3 has been changed to python3.7 in the bridge file under co-simulation scenario folder.